pkg/controller/appmetricsservice/appmetricsservice_controller.go

```diff
@@ -332,6 +396,60 @@ func (r *ReconcileAppMetricsService) Reconcile(request reconcile.Request) (recon
 	}
 	//#endregion
 
+	//#region Backups
+	if len(instance.Spec.Backups) > 0 {
+		backupjobSA := &corev1.ServiceAccount{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: "backupjob", Namespace: instance.Namespace}, backupjobSA)
+		if err != nil {
+			reqLogger.Error(err, "A 'backupjob' ServiceAccount is required for the requested backup CronJob(s). Will check again in 10 seconds")
+			return reconcile.Result{RequeueAfter: time.Second * 10}, nil
+		}
+	}
+
+	existingCronJobs := &batchv1beta1.CronJobList{}
+	opts = client.InNamespace(instance.Namespace).MatchingLabels(labels(instance, "backup"))
+	err = r.client.List(context.TODO(), opts, existingCronJobs)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	desiredCronJobs, err := backups(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	for _, desiredCronJob := range desiredCronJobs {
+		if err := controllerutil.SetControllerReference(instance, &desiredCronJob, r.scheme); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if exists := containsCronJob(existingCronJobs.Items, &desiredCronJob); exists {
```

Comment:
Why not impl as it is done in the Getting Started examples? See. https://github.com/operator-framework/operator-sdk-samples/blob/master/memcached-operator/pkg/controller/memcached/memcached_controller.go#L105

Reply:
Also, see here for example: https://github.com/aerogear/mobile-security-service-operator/blob/master/pkg/controller/mobilesecurityservicebackup/controller.go#L239

Reply:
This is a more complex case than either of the examples you list. Here we've got a list of backup specs provided from the CR, and we want to reconcile that with the list of cronjobs in the cluster. If there are changes to the backups in the CR (could be added/removed/updated) we want those changes to be reflected in the cluster.

Reply:
@grdryn shows to me that it can work in the same way. The only diff is that you will do a to create a cronjob for each database pod.  Why not add the owner when the object is created?

Reply:
I think you're misunderstanding. It's not about the number of database server pods, it's about backup specs. These wouldn't necessarily scale with database pods, but maybe with different schedules or backup locations or encryption targets (or additional bits of state that need to be backed up in future, for example if we had a redis service provisioned here, then we could have one (or more) backup for postgresql and one (or more) for redis or whatever.

Reply:
Hi @grdryn,

I understood now what you would like to achieve with. However, you do not need to use any for at all. Note that the only thing that you need is to create the cronjob object with the name of the CR. 
The name of the CR is a unique identifier by namespace. 

For example, see here: https://github.com/aerogear/mobile-security-service-operator/blob/master/pkg/controller/mobilesecurityserviceapp/controller.go#L219
In the MSS Oper, we create a configMap for each APP CR applied in the namespace and NO loops are required at all. Also, what you would like to achieve here ( apply N BKP CRS in the same namespace to create N cronjobs ) is already supported by the BKP in MSS Oper as well. See here: https://github.com/aerogear/mobile-security-service-operator/tree/master/pkg/controller/mobilesecurityservicebackup. Also, I'd like to recommend you check its CR's to have a better idea overall that it is able to do: https://github.com/aerogear/mobile-security-service-operator/blob/master/deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicebackup_cr.yaml

Reply:
We're going with a different approach here than in the mobile-security-service-operator. We're not going to expose a "backup CRD" and have one CR per backup that we want. Instead we're going with an optional list of backup configurations in the service CR, and therefore have this loop.